### PR TITLE
Adding PHPStan

### LINKED ISSUE #135: Delete .phpcs cache directory when rolling up to parent
# Issue

When the "rollup phpcs to parent" option is selected, the `.phpcs` directory with the `.gitkeep` file is maintained, even though the configuration should be handled by the parent, and assumptions shouldn't be made about the location of the cache directory.

# Steps to Reproduce

Create a new WordPress plugin and select the option to rollup phpcs configuration to the parent. Observe that the `.phpcs` directory still exists in the plugin folder.

# Recommended Solution

Remove the `.phpcs` directory and its contents from the plugin if "rollup phpcs to parent" option is selected.

### PR DESCRIPTION
**Pull request by srtfisher**
Adds PHPStan set to the maximum level to help with code debugging.

Also fixes #135

### CODE REVIEW

File: .github/workflows/coding-quality.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  coding-quality:
```
Comment (renatonascalves) [inline .github/workflows/coding-quality.yml]:
```suggestion
  code-quality:
```

Comment (renatonascalves) [reply .github/workflows/coding-quality.yml]: Something I forgot to update in the original repo.

File: composer.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "require-dev": {
         "alleyinteractive/alley-coding-standards": "^1.0",
         "mantle-framework/testkit": "^0.10",
-        "nunomaduro/collision": "^5.0"
+        "nunomaduro/collision": "^5.0",
```
Comment (renatonascalves) [inline composer.json]: Is this required? Since `@mantle-framework/testkit` already bundles it...

**Review by renatonascalves — APPROVED** [review]
🗡️

File: composer.json (hunk 2 above)
Comment (szepeviktor) [reply composer.json]: https://github.com/mantle-framework/testing/blob/709076ab3e61668da048e343324c027d0737ce66/composer.json#L15

Comment (srtfisher) [reply composer.json]: Removed!

**Pull request merged**